src/components/characters/CardContainer.jsx

```diff
@@ -1,137 +1,79 @@
-import React, { useState, useEffect, useCallback } from 'react';
+import React from 'react';
+import useSWRInfinite from 'swr/infinite';
+import { Button, Box, Typography, CircularProgress } from '@mui/material';
 import Cards from './Cards';
-import { Button, Container, Typography, CircularProgress } from '@mui/material';
+import ErrorDisplay from './ErrorDisplay';
+import LoadingDisplay from './LoadingDisplay';
 
-const API_URL = 'https://rickandmortyapi.com/api/character/?page=1';
+import { commonStyles, CHARACTER_API_URL } from '../../constants';
+import { headerStyles } from './typographyStyles';
 
-const CardContainer = () => {
-  const [characters, setCharacters] = useState([]);
-  const [loading, setLoading] = useState(true);
-  const [error, setError] = useState(null);
-  const [totalCount, setTotalCount] = useState(null);
-  const [nextPageUrl, setNextPageUrl] = useState(API_URL);
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
-  const fetchCharacters = useCallback(async () => {
-    if (!nextPageUrl) {
-      return;
-    }
-    setLoading(true);
-    try {
-      const response = await fetch(nextPageUrl);
-      if (!response.ok) {
-        throw new Error(
-          `Network response was not ok! Status: ${response.status}`,
-        );
-      }
-      const {
-        info: { next: newNext, count },
-        results,
-      } = await response.json();
-      // setTimeout(() => {
-      setCharacters((prevCharacters) => [...prevCharacters, ...results]);
-      if (!totalCount) {
-        setTotalCount(count);
-        console.log('Total count ' + count);
-      }
-      setNextPageUrl(newNext);
-      setLoading(false);
-      // }, 1000);
-    } catch (error) {
-      console.error('Fetch error: ', error);
-      setError(error);
-      setLoading(false);
-    }
-  }, [nextPageUrl]);
+const CardContainer = () => {
+  const getKey = (pageIndex, prevCharacters) => {
```

Comment:
[nth] if there is an parameter which you have to add but don't use, the underscore symbol is used instead of a name ([more](https://stackoverflow.com/a/27637038))
```suggestion
  const getKey = (_, prevCharacters) => {
```